be/src/olap/tuple_reader.cpp

```diff
@@ -186,15 +186,24 @@ Status TupleReader::_unique_key_next_row(RowCursor* row_cursor, MemPool* mem_poo
         // in UNIQUE_KEY highest version is the final result, there is no need to
         // merge the lower versions
         direct_copy_row(row_cursor, *_next_key);
-        // skip the lower version rows;
-        auto res = _collect_iter.next(&_next_key, &_next_delete_flag);
-        if (LIKELY(res.precise_code() != OLAP_ERR_DATA_EOF)) {
-            if (UNLIKELY(!res.ok())) {
-                LOG(WARNING) << "next failed: " << res;
-                return res;
+        while (_next_key) {
+            // skip the lower version rows;
+            auto res = _collect_iter.next(&_next_key, &_next_delete_flag);
+            if (LIKELY(res.precise_code() != OLAP_ERR_DATA_EOF)) {
+                if (UNLIKELY(!res.ok())) {
+                    LOG(WARNING) << "next failed: " << res;
+                    return res;
+                }
+
+                if (!equal_row(_key_cids, *row_cursor, *_next_key)) {
```

Comment:
@HappenLee Why do we need to merge keys here?
1. If there are duplicate keys in one rowset, the MergeIterator will merge them.
2. If there are duplicate keys between rowsets, the CollectIterator will merge them.